### PR TITLE
updated the certs tests to use host class

### PR DESCRIPTION
**Pull request by omkarkhatavkar**
```
============================= test session starts ==============================
platform linux -- Python 3.8.6, pytest-6.2.3, py-1.10.0, pluggy-0.13.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/okhatavk/okhatavk/Satellite/robottelo, configfile: pyproject.toml
plugins: services-2.2.1, cov-2.10.1, xdist-2.2.1, reportportal-5.0.8, mock-3.5.1, forked-1.3.0, ibutsu-1.14.1, picked-0.4.6collected 11 items / 2 deselected / 9 selected

../../okhatavk/Satellite/robottelo/tests/foreman/sys/test_katello_certs_check.py  [ 11%]                                                                 [100%]

=============================== warnings summary ===============================

```

### CODE REVIEW

**Comment by omkarkhatavkar** [comment]
trigger: test-robottelo
pytest: tests/foreman/sys/test_katello_certs_check.py

**Comment by omkarkhatavkar** [comment]
trigger: test-robottelo
pytest: tests/foreman/sys/test_katello_certs_check.py

**Comment by omkarkhatavkar** [comment]
trigger: test-robottelo
pytest: tests/foreman/sys/test_katello_certs_check.py

File: tests/foreman/sys/test_katello_certs_check.py [hunk 1]
```diff
@@ -529,32 +455,27 @@ class TestCapsuleCertsCheckTestCase:
     on both Satellite Server's base system and on local host.
     """
 
-    @pytest.fixture(scope="module")
-    def file_setup(self):
+    @pytest.fixture
+    def capsule_certs_teardown(self, default_sat):
         """Create working directory and file."""
-        capsule_hostname = 'capsule.example.com'
+        capsule = AttrDict({'hostname': 'capsule.example.com'})
```
[the remark this answers is not in the document]
Comment (JacobCallahan) [reply tests/foreman/sys/test_katello_certs_check.py:457]: haha no, this works

File: robottelo/hosts.py [hunk 2]
```diff
@@ -1056,19 +1057,54 @@ def url_katello_ca_rpm(self):
...
+            'generate-crt.sh',
+            'openssl.cnf',
+            'certs.sh',
+            'extensions.txt',
+        ]:
+            self.session.sftp_write(get_data_file(file), f'/root/{file}')
+        self.execute('echo 100001 > serial')
+        self.execute('bash generate-ca.sh')
+        self.execute(f'yes | bash generate-crt.sh {self.hostname}')
+        self.execute('bash certs.sh')
```
Comment (JacobCallahan) [inline robottelo/hosts.py:1087]: Can you either check the exit code or some state on the system to validate whether this method did in fact do everything it was support to before returning?

File: tests/foreman/sys/test_katello_certs_check.py [hunk 3]
```diff
@@ -307,45 +241,40 @@ def test_positive_update_katello_certs(self, cert_setup, cert_data, certs_cleanu
...
-                # assert no hammer ping SSL cert error
-                result = connection.run('hammer ping')
-                assert 'SSL certificate verification failed' not in result.stdout
-                assert 'ok' in result.stdout[1]
-                # assert all services are running
-                result = connection.run('foreman-maintain health check --label services-up -y')
-                assert result.return_code == 0, 'Not all services are running'
+            cert_data, satellite = cert_setup_destructive_teardown
+            result = satellite.execute('hammer ping')
+            assert result.status == 0, 'Hammer Ping fail'
```
Comment (JacobCallahan) [inline tests/foreman/sys/test_katello_certs_check.py:246]:
```suggestion
            assert result.status == 0, f'Hammer Ping failed:\n{result.stderr}'
```

**Review by JacobCallahan — CHANGES_REQUESTED** [review]
small changes

**Review by JacobCallahan — APPROVED** [review]
ACK, thanks!

**Comment by omkarkhatavkar** [comment]
trigger: test-robottelo
pytest: tests/foreman/sys/test_katello_certs_check.py

**Review by swadeley — APPROVED** [review]
ACK

**Comment by omkarkhatavkar** [comment]
trigger: test-robottelo
pytest: tests/foreman/sys/test_katello_certs_check.py

**Comment by omkarkhatavkar** [comment]
trigger: test-robottelo
pytest: tests/foreman/sys/test_katello_certs_check.py

**Comment by omkarkhatavkar** [comment]
trigger: test-robottelo
pytest: tests/foreman/sys/test_katello_certs_check.py

**Comment by omkarkhatavkar** [comment]
trigger: test-robottelo
pytest: tests/foreman/sys/test_katello_certs_check.py

**Comment by omkarkhatavkar** [comment]
trigger: test-robottelo
pytest: tests/foreman/sys/test_katello_certs_check.py

**Review by mshriver — APPROVED** [review]

**Pull request merged**